src/Ocelot/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -17,15 +18,17 @@ public static class ServiceCollectionExtensions
     /// <returns>An <see cref="IOcelotBuilder"/> object.</returns>
     public static IOcelotBuilder AddOcelot(this IServiceCollection services)
     {
-        var configuration = services.BuildServiceProvider().GetRequiredService<IConfiguration>();
+        var descriptor = services.Where(descriptor => descriptor.ServiceType == typeof(IConfiguration)).FirstOrDefault();
```

Comment:
Not an issue, but my questions:
- How many `IConfiguration` objects can be registered in DI? If more than one instance, how to detect exactly `ocelot.json` one?
- Could not the `IConfiguration` be registered at all? Seems not, because of `appsettings.json`...

Reply:
`IConfiguration` is a singleton. When adding multiple configuration sources, these are mixed together as a single configuration context, where the latest source overrides former ones in case of configuration key duplication.

Technically you could provoke an error if you provide a `new ServiceCollection()`, but these methods are designed to be used with `WebApplicationBuilder` which does formally creates an IConfiguration instance.

Perhpas this could cause problems with slimmer versions of the builder introduced in dotnet 8.

Reply:
By flattening multiple sources in a single IConfiguration context, the framework allows to easily implement "configuration overriding" features. For instance the default WebApplicationBuilder registers appsettings.json, then user secret file, and then environment variables as configuration sources. This allows secret files to override appsettings.json and environment variables to override both.

Reply:
Also using Builder without an IConfiguration allows to register configuration sources AFTER the registration of the services. This in turns allow to do some fancy stuff like configuration bootstraping (read configuration from file, create services to access a remote storage using this config, then provide the rest of the app with config values from that storage).

Which is exacly the kind of problems Ocelot had tried to solve with Consul/Eureka.

Reply:
If you want to isloate `ocelot.json`, you need to have a separate IConfiguration instance, using a separate ConfigurationBuilder for instance and not have it injected in the DI container. This is NOT what currently happens and NEITHER is what is described in the documentation.

Currently all configuration (ocelot routes and aspnet config) are mixed together in a single IConfiguration